k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala

```diff
@@ -352,12 +402,21 @@ class IngressCacheTest extends FunSuite with Awaits {
   val annotationClass = "linkerd"
 
   def mkIngressApiServiceReturning(response: String) = Service.mk[Request, Response] {
+    mkWatchedIngressApiServiceReturning(response, "")
+  }
+
+  def mkWatchedIngressApiServiceReturning(response: String, watchResponse: String) = Service.mk[Request, Response] {
     case req if req.uri.contains("/apis/extensions/v1beta1/ingresses") =>
       val rsp = Response()
       rsp.content = Buf.Utf8(response)
       Future.value(rsp)
     case req if req.uri.contains("/apis/extensions/v1beta1/watch/ingresses") =>
-      Future.value(Response())
+      var doWatch = new Promise[Unit]
+      doWatch.setDone()
```

Comment:
In this case, I think the `doWatch` promise is unnecessary --- in `EndpointsNamerTest`, we use those `Promise` to allow the test body to signal to the mock API service when a particular event should be sent, but here you immediately set the `doWatch` promise to be done inside of the mock API service. Since `doWatch` is done, the line
```scala
doWatch before rsp.writer.write(Buf.Utf8(watchResponse))
```
should be equivalent to just
```scala
rsp.writer.write(Buf.Utf8(watchResponse))
```
so I think you can remove the `doWatch` promise entirely.

Reply:
I could have _sworn_ this test only started working as expected when I added the Promise, but it works totally fine without them as you say. Thanks!